pkg/controller/plan/adapter/openstack/builder.go

```diff
@@ -46,32 +243,262 @@ func (r *Builder) VirtualMachine(vmRef ref.Ref, object *cnv.VirtualMachineSpec,
+		cpuPolicy = flavorCPUPolicy
+	} else if imageCPUPolicy, ok := vm.Image.Properties[CpuPolicy]; ok {
+		cpuPolicy = imageCPUPolicy.(string)
+	}
+
+	if cpuPolicy == CpuPolicyDedicated {
+		object.Template.Spec.Domain.CPU.DedicatedCPUPlacement = true
+	}
+
+	if flavorEmulatorThreadPolicy, ok := vm.Flavor.ExtraSpecs[FlavorEmulatorThreadPolicy]; ok {
+		if flavorEmulatorThreadPolicy == CpuThreadPolicyIsolate {
+			object.Template.Spec.Domain.CPU.IsolateEmulatorThread = true
+		}
+	}
+
+	// Set CPU Sockets/Cores/Threads and Memory requests
+	// TODO support NUMA, CPU pinning
+	var cpuSockets, cpuCores, cpuThreads int
+
+	cpuSockets = vm.Flavor.VCPUs
+	cpuCores = 1
+	cpuThreads = 1
+
+	if flavorCPUSockets, ok := vm.Flavor.ExtraSpecs[FlavorCpuSockets]; ok {
+		if sockets, err := strconv.Atoi(flavorCPUSockets); err == nil {
+			cpuSockets = sockets
+		}
+	} else if imageCPUSockets, ok := vm.Image.Properties[CpuSockets]; ok {
+		if sockets, ok := imageCPUSockets.(int); ok {
+			cpuSockets = sockets
+		}
+	}
+	if flavorCPUCores, ok := vm.Flavor.ExtraSpecs[FlavorCpuCores]; ok {
+		if cores, err := strconv.Atoi(flavorCPUCores); err == nil {
+			cpuCores = cores
+		}
+	} else if imageCPUCores, ok := vm.Image.Properties[CpuCores]; ok {
+		if cores, ok := imageCPUCores.(int); ok {
+			cpuCores = cores
+		}
+	}
+	if flavorCPUThreads, ok := vm.Flavor.ExtraSpecs[FlavorCpuThreads]; ok {
+		if threads, err := strconv.Atoi(flavorCPUThreads); err == nil {
+			cpuThreads = threads
+		}
+	} else if imageCPUThreads, ok := vm.Image.Properties[CpuThreads]; ok {
+		if threads, ok := imageCPUThreads.(int); ok {
+			cpuThreads = threads
+		}
+	}
+
+	resourceRequests := map[core.ResourceName]resource.Quantity{}
+
+	object.Template.Spec.Domain.CPU.Sockets = uint32(cpuSockets)
+	object.Template.Spec.Domain.CPU.Cores = uint32(cpuCores)
+	object.Template.Spec.Domain.CPU.Threads = uint32(cpuThreads)
+
+	// TODO Support HugePages
+	memory := resource.NewQuantity(int64(vm.Flavor.RAM)*1024*1024, resource.BinarySI)
+	resourceRequests[core.ResourceMemory] = *memory
+
+	object.Template.Spec.Domain.Resources.Requests = resourceRequests
+}
+
 func (r *Builder) mapDisks(vm *model.Workload, persistentVolumeClaims []core.PersistentVolumeClaim, object *cnv.VirtualMachineSpec) {
 	var kVolumes []cnv.Volume
 	var kDisks []cnv.Disk
 
 	pvcMap := make(map[string]*core.PersistentVolumeClaim)
 
+	// The disk bus is common for all the VM disks and it's configured in the image properties.
+	bus := DefaultProperties[DiskBus]
+	if imageDiskBus, ok := vm.Image.Properties[DiskBus]; ok {
+		bus = imageDiskBus.(string)
+	} else {
+		for _, volume := range vm.Volumes {
+			if volume.Bootable == "true" {
+				if volumeDiskBus, ok := volume.VolumeImageMetadata[DiskBus]; ok {
+					bus = volumeDiskBus
+				}
+			}
+		}
+	}
+	// Only q35 machine type is supported in Kubevirt so we need to map
+	// openstack bus types to supported ones
+	switch bus {
+	case IdeBus:
+		bus = SataBus
+	case ScsiBus:
```

Comment:
so you mean IDE should be mapped to SATA and everything else besides SCSI should be mapped to VIRTIO? if so, how about specifying the SCSI->SCSI explicitly, for readability?

Reply:
or add `if bus != ScsiBus` below